base_local_planner/src/costmap_model.cpp

```diff
@@ -59,13 +59,15 @@ namespace base_local_planner {
     if(!costmap_.worldToMap(position.x, position.y, cell_x, cell_y))
       return -3.0;
 
+    //check its cost, so we can skip footprint check if it's already non-traversable space
+    unsigned char cost = costmap_.getCost(cell_x, cell_y);
+    if(cost == NO_INFORMATION)
+      return -2.0;
```

Comment:
Q: should we check for lethal if center is `NO_INFORMATION`?
I think `NO_INFORMATION` may be ignored for some controllers.
For instance, in PF we have a flag to ignore it, and I think that is what we are doing.
Besides the misguiding value of the cost (255 and 254), I think `LETHAL` is a hard constraint, and `NO_INFORMATION` is a soft one.

Reply:
Could be but anyway this doesn't change the logic we use now, namely return the first negative cost you find, without caring about precedence  (that is, if we find a NO_INFORMATION when checking the footprint, we return it, without searching for a LETHAL)

Reply:
true!